modules/data-streams/src/internalClusterTest/java/org/elasticsearch/datastreams/TSDBPassthroughIndexingIT.java

```diff
@@ -187,6 +198,89 @@ public void testIndexingGettingAndSearching() throws Exception {
+                .dataStreamTemplate(new ComposableIndexTemplate.DataStreamTemplate(false, false))
+                .build()
+        );
+        client().execute(TransportPutComposableIndexTemplateAction.TYPE, request).actionGet();
+
+        Instant time = Instant.now();
+        int numBulkItems = randomIntBetween(16, 128);
+        var bulkRequest = new BulkRequest(dataStreamName);
+        for (int i = 0; i < numBulkItems; i++) {
+            var indexRequest = new IndexRequest(dataStreamName).opType(DocWriteRequest.OpType.CREATE);
+            indexRequest.source(
+                DOC.replace("$time", formatInstant(time))
+                    .replace("$uid", randomUUID())
+                    .replace("$name", randomAlphaOfLength(4))
+                    .replace("$ip", InetAddresses.toAddrString(randomIp(randomBoolean()))),
+                XContentType.JSON
+            );
+            bulkRequest.add(indexRequest);
+            time = time.plusMillis(1);
+        }
+
+        bulkRequest.setRefreshPolicy(WriteRequest.RefreshPolicy.WAIT_UNTIL);
+        var bulkResponse = client().bulk(bulkRequest).actionGet();
+        for (var itemResponse : bulkResponse) {
+            String id = itemResponse.getId();
+            String index = itemResponse.getIndex();
+            var getResponse = client().get(new GetRequest(index, id)).actionGet();
+            assertThat(getResponse.isExists(), is(true));
+
+            var searchRequest = new SearchRequest(index);
+            searchRequest.source(new SearchSourceBuilder().query(new TermQueryBuilder("_id", id)));
+            assertResponse(client().search(searchRequest), searchResponse -> {
+                assertHitCount(searchResponse, 1);
+                assertThat(searchResponse.getHits().getHits()[0].getId(), equalTo(id));
+            });
+        }
+
+        var rolloverResponse = client().admin().indices().rolloverIndex(new RolloverRequest(dataStreamName, null)).actionGet();
+        assertThat(rolloverResponse.isRolledOver(), is(true));
+        String sourceIndex = rolloverResponse.getOldIndex();
+
+        var updateSettingsResponse = client().admin()
+            .indices()
+            .updateSettings(new UpdateSettingsRequest(sourceIndex).settings(Settings.builder().put("index.blocks.write", true)))
+            .actionGet();
+        assertThat(updateSettingsResponse.isAcknowledged(), is(true));
+
+        String shrunkenTarget = "k8s-shrunken";
+        var shrinkIndexResponse = client().admin()
+            .indices()
+            .prepareResizeIndex(sourceIndex, shrunkenTarget)
+            .setResizeType(ResizeType.SHRINK)
+            .setSettings(indexSettings(2, 0).build())
+            .get();
+        assertThat(shrinkIndexResponse.isAcknowledged(), is(true));
+        assertThat(shrinkIndexResponse.index(), equalTo(shrunkenTarget));
+
+        for (var itemResponse : bulkResponse) {
+            String id = itemResponse.getId();
```

Comment:
So, we want the ids before and after shrinking to be stable? Is this required?

Reply:
Yes, the ids should be the same before and after shrinking.
With the current document based APIs this is required.